### PR TITLE
release: v0.29.0

### PR DESCRIPTION
**Pull request by djm81**
# Description

Release PR to merge `dev` into `main` for **v0.29.0**.

This release includes module lifecycle management and init UX improvements, including:
- split init flow (`init` baseline vs `init ide` for prompt/template setup)
- interactive/non-interactive module enable/disable behavior improvements
- dependency-aware module lifecycle behavior (force mode propagation)
- related test and typing fixes from follow-up CI findings

**Fixes** #N/A

**New Features** #N/A

**Contract References**: No new public runtime contract surface introduced beyond existing module lifecycle behavior changes.

## Type of Change

Please check all that apply:

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Contract enforcement (adding/updating `@icontract` decorators)
- [x] 🧪 Test enhancement (scenario tests, property-based tests)
- [x] 🔧 Refactoring (code improvement without functionality change)

## Contract-First Testing Evidence

**Required for all changes affecting CLI commands or public APIs:**

### Contract Validation

- [ ] **Runtime contracts added/updated** (`@icontract` decorators on public APIs)
- [ ] **Type checking enforced** (`@beartype` decorators applied)
- [ ] **CrossHair exploration** completed: `hatch run contract-test-exploration`
- [ ] **Contract violations** reviewed and addressed

### Test Execution

- [ ] **Contract validation**: `hatch run contract-test-contracts` ✅
- [ ] **Contract exploration**: `hatch run contract-test-exploration` ✅
- [ ] **Scenario tests**: `hatch run contract-test-scenarios` ✅
- [ ] **Full test suite**: `hatch run contract-test-full` ✅

### Test Quality

- [x] **CLI commands tested** with typer test client
- [ ] **Edge cases covered** with Hypothesis property tests
- [x] **Error handling tested** with invalid inputs
- [x] **Rich console output verified** manually or with snapshots

## How Has This Been Tested?

**Contract-First Approach**: Module lifecycle behavior validated via unit/integration tests and CI quality gates from `dev` branch.

### Manual Testing

- [x] Tested CLI commands manually
- [x] Verified rich console output
- [x] Tested with different input scenarios
- [x] Checked error messages for clarity

### Automated Testing

- [x] Contract validation passes
- [ ] Property-based tests cover edge cases
- [x] Scenario tests cover user workflows
- [x] All existing tests still pass

### Test Environment

- Python version: 3.12
- OS: Ubuntu (CI) and local Linux

## Checklist

- [x] My code follows the style guidelines (PEP 8, ruff format, isort)
- [x] I have performed a self-review of my code
- [ ] I have added/updated contracts (`@icontract`, `@beartype`)
- [x] I have added/updated docstrings (Google style)
- [x] I have made corresponding changes to documentation
- [x] My changes generate no new warnings (basedpyright, ruff, pylint)
- [x] All tests pass locally
- [x] I have added tests that prove my fix/feature works
- [x] Any dependent changes have been merged

## Quality Gates Status

- [x] **Type checking** ✅ (`hatch run type-check`)
- [ ] **Linting** ✅ (`hatch run lint`)
- [ ] **Contract validation** ✅ (`hatch run contract-test-contracts`)
- [ ] **Contract exploration** ✅ (`hatch run contract-test-exploration`)
- [ ] **Scenario tests** ✅ (`hatch run contract-test-scenarios`)

## Screenshots/Recordings (if applicable)

N/A.

### CODE REVIEW

File: src/specfact_cli/modules/init/src/commands.py [hunk 1]
```diff
@@ -111,21 +125,313 @@ def _copy_backlog_field_mapping_templates(repo_path: Path, force: bool, console:
...
+        return dev_templates_dir
+    try:
+        import importlib.resources
+
+        resources_ref = importlib.resources.files("specfact_cli")
+        templates_ref = resources_ref / "resources" / "prompts"
+        package_templates_dir = Path(str(templates_ref)).resolve()
+        if package_templates_dir.exists():
+            return package_templates_dir
+    except Exception:
```
[the remark this answers is not in the document]
Comment (djm81) [reply src/specfact_cli/modules/init/src/commands.py]: Fixed.
